core/src/OpenMP/Kokkos_OpenMP_Task.cpp

```diff
@@ -75,9 +75,22 @@ HostThreadTeamDataSingleton::HostThreadTeamDataSingleton()
       num_pool_reduce_bytes, num_team_reduce_bytes, num_team_shared_bytes,
       num_thread_local_bytes);
 
+  void* ptr = nullptr;
+  try {
+    ptr = space.allocate(alloc_bytes);
+  } catch (Experimental::RawMemoryAllocationFailure const& f) {
+    // For now, just rethrow the error message with a note
+    // Note that this could, in turn, trigger an out of memory exception,
+    // but it's pretty unlikely, so we won't worry about it for now.
+    // TODO reasonable error message when `std::string` causes OOM error
```

Comment:
The rest of these are careful not to do allocation in the error pathway; see `RawAllocationError::what()` for instance. The reason we should care is because it means that users who are actually running out of memory won’t see a reasonable error message. Also, users who are theoretically catching these errors (in order to, say, clear a cache of some sort and try again, which is a thing people do occasionally in HPC) won’t catch the right one.